src/shell/layout/tiling/grabs/resize.rs

```diff
@@ -184,9 +186,11 @@ pub struct ResizeForkGrab {
     last_loc: Point<f64, Global>,
     old_tree: Option<Tree<Data>>,
     accumulated_delta: f64,
+    accumulated_delta_parent: f64,
     node: NodeId,
     output: WeakOutput,
     left_up_idx: usize,
+    parent_left_up_idx: Option<usize>,
```

Comment:
Instead of splitting this in `parent` and `node`/<no suffix>, I would prefer, if we separately determine the `left_idx` and `left_node` and the `up_idx` and the `up_node`. (And have both be an `Option`, though the grab should obviously never be created with both being `None`.)

Reply:
The `*_node` values would be instances of something like `enum ResizeDest { Node, NodeParent }`?

Reply:
Okay, looking at your other comment about having to walk up the tree until finding a split of the correct orientation, this needs to be a `ResizeDest(usize)` meaning how many times we need to walk towards the root?

Reply:
Okay, looking at your other comment about having to walk up the tree until finding a split of the correct orientation, this needs to be a `ResizeDest(usize)` meaning how many times we need to walk towards the root?

Reply:
> 

The *_node values would be instances of something like enum ResizeDest { Node, NodeParent }?


No just `NodeId`s. You don't need to know how far you have walked, if you just mark the nodes we do the resize operations on. (Both of them one for each axis.)